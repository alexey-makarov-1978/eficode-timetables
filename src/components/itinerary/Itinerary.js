import React from "react";
import PropTypes from "prop-types";
import Leg from "../leg/Leg";
import { getTime, getDuration } from "../utils/utils";

import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";
import DirectionsBusIcon from "@material-ui/icons/DirectionsBus";
import Accordion from "@material-ui/core/Accordion";
import AccordionDetails from "@material-ui/core/AccordionDetails";
import AccordionSummary from "@material-ui/core/AccordionSummary";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";

const Itinerary = ({ data }) => {
  const classes = useStyles();

  return (
    <div>
      <Accordion>
        <AccordionSummary expandIcon={<ExpandMoreIcon />}>
          <div className={classes.grid}>
            <Grid container>
              <Grid item xs={2}>
                <b>{getTime(data.startTime)}</b>
              </Grid>
              <Grid item xs={4}>
                <MoreHorizIcon />
                <DirectionsBusIcon />
                <MoreHorizIcon />
              </Grid>
              <Grid item xs={6}>
                <b>
                  {getTime(data.endTime)} ({getDuration(data.duration)})
                </b>
              </Grid>
            </Grid>
          </div>
        </AccordionSummary>
        <AccordionDetails>
          <Grid container>
            {data.legs.map((leg, index) => (
              <Grid key={index} item xs={12}>
                <Leg data={leg} />
              </Grid>
            ))}
          </Grid>
        </AccordionDetails>
      </Accordion>
    </div>
  );
};

const useStyles = makeStyles((theme) => ({
  grid: {
    maxWidth: "100%",
    flexGrow: 1,
    padding: 2,
  },
}));

Itinerary.propTypes = {
  data: PropTypes.object.isRequired,
};

export default Itinerary;
