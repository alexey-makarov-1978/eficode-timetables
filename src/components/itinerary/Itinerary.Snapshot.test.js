import React from "react";
import Itinerary from "./Itinerary";
import renderer from "react-test-renderer";

it("Itinerary matching snapshot with 3 legs", () => {
  const itinerary = {
    startTime: 1598869030000,
    endTime: 1598869904000,
    duration: 874,
    legs: [
      {
        startTime: 1598869030000,
        endTime: 1598869261000,
        mode: "WALK",
        duration: 231.0,
        distance: 261.037,
        route: null,
        from: {
          name: "Origin",
        },
        to: {
          name: "Hanken",
        },
      },
      {
        startTime: 1598869261000,
        endTime: 1598869715000,
        mode: "TRAM",
        duration: 454.0,
        distance: 1733.578241151397,
        route: {
          shortName: "1",
        },
        from: {
          name: "Hanken",
        },
        to: {
          name: "Kaupunginpuutarha",
        },
      },
      {
        startTime: 1598869715000,
        endTime: 1598869904000,
        mode: "WALK",
        duration: 189.0,
        distance: 204.855,
        route: null,
        from: {
          name: "Kaupunginpuutarha",
        },
        to: {
          name: "Destination",
        },
      },
    ],
  };

  const tree = renderer.create(<Itinerary data={itinerary} />);
  expect(tree).toMatchSnapshot();
});
