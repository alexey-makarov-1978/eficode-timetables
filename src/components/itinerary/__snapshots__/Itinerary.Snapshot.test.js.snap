// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary matching snapshot with 3 legs 1`] = `
<div>
  <div
    className="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content"
      >
        <div
          className="makeStyles-grid-1"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <b>
                13:17
              </b>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 16c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4s-8 .5-8 4v10zm3.5 1c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm9 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6H6V6h12v5z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <b>
                13:31
                 (
                14 min
                )
              </b>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiAccordionDetails-root"
            >
              <div
                className="MuiGrid-root MuiGrid-container"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="makeStyles-root-2"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <b>
                          13:17
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        <b>
                          Origin
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      />
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        WALK 261 m (3 min)
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <b>
                          13:21
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        <b>
                          Hanken
                        </b>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="makeStyles-root-2"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <b>
                          13:21
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        <b>
                          Hanken
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      />
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        TRAM (1) 1 km (7 min)
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <b>
                          13:28
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        <b>
                          Kaupunginpuutarha
                        </b>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="makeStyles-root-2"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <b>
                          13:28
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        <b>
                          Kaupunginpuutarha
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      />
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        WALK 204 m (3 min)
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <b>
                          13:31
                        </b>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
                          />
                        </svg>
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
                      >
                        <b>
                          Destination
                        </b>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
